src/import-data/protected-data-utils.ts

```diff
@@ -16,7 +16,7 @@ export const isProtectedSource = (url: string, filetype: string): boolean => {
       'gen3-biodatacatalyst-nhlbi-nih-gov-pfb-export.s3.amazonaws.com',
       'gen3-theanvil-io-pfb-export.s3.amazonaws.com',
     ];
-    return cond([!filetype || !url, () => false], [filetype.toLowerCase() !== 'pfb', () => false], () =>
+    return cond([!filetype || !url, () => false], [!!filetype && filetype.toLowerCase() !== 'pfb', () => false], () =>
```

Comment:
We know `url` is defined if we reach here so I think `!url` is unnecessary.

Reply:
TypeScript does not understand that though. Alternatively, this could use a non-null assertion.

```suggestion
    return cond([!filetype || !url, () => false], [filetype!.toLowerCase() !== 'pfb', () => false], () =>
```

Reply:
My comment was about the `if (!url)` check on line 7. If we don't return on line 8, then we know that `url` is defined and checking it again here has no effect. Unless I'm missing something. As it is, `url` isn't used in this block anyway so it doesn't matter if it's defined or not, as far as I can tell.

Reply:
Oh, I see. Yeah, you're right. The `|| !url` check in the first case is superfluous.

Reply:
This is cleaned up in #4305.